assets/javascripts/discourse/widgets/alert-receiver.js.es6

```diff
@@ -152,34 +152,94 @@ createWidget("alert-receiver-row", {
   tagName: "tr",
 
   transform(attrs) {
+    const starts_at = attrs.alert.starts_at;
+    const ends_at = attrs.alert.ends_at || new Date().toISOString();
+
+    let linkText = this.attrs.alert.link_text;
+    if (linkText === "") {
+      linkText = null;
+    }
+    linkText = linkText || I18n.t("prom_alert_receiver.actions.link");
+
     return {
-      logsUrl: this.buildLogsUrl(attrs),
-      graphUrl: this.buildGraphUrl(attrs),
-      grafanaUrl: this.buildGrafanaUrl(attrs),
+      generatorUrl: this.processUrl(
+        attrs.alert.generator_url,
+        starts_at,
+        ends_at
+      ),
+      linkUrl: this.processUrl(attrs.alert.link_url, starts_at, ends_at),
+      linkText,
     };
   },
 
-  buildLogsUrl(attrs) {
-    const base = attrs.alert.logs_url;
-    if (!base) {
+  processUrl(urlString, startsAt, endsAt) {
+    try {
+      const url = new URL(urlString);
+
+      return (
+        this.buildPrometheusUrl(url, startsAt, endsAt) ||
+        this.buildGrafanaUrl(url, startsAt, endsAt) ||
+        this.buildKibanaUrl(url, startsAt, endsAt) ||
+        url
+      );
+    } catch (e) {
+      if (e instanceof TypeError) {
+        // Invalid or blank URL
+        return;
+      }
+      throw e;
+    }
+  },
+
+  checkMatch(url, regexString) {
+    if (!regexString || !url) {
+      return false;
+    }
+    if (regexString === "" || url === "") {
+      return false;
+    }
+
+    try {
+      const regexp = new RegExp(regexString);
+      return regexp.test(url);
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error("Alert receiver regex error", e);
+    }
+
+    return false;
+  },
+
+  buildKibanaUrl(url, starts_at, ends_at) {
+    const regex = this.siteSettings.prometheus_alert_receiver_kibana_regex;
+    if (!this.checkMatch(url, regex)) {
       return;
     }
-    const start = attrs.alert.starts_at;
-    const end = attrs.alert.ends_at || new Date().toISOString();
-    return `${base}#/discover?_g=(time:(from:'${start}',mode:absolute,to:'${end}'))`;
+
+    // Kibana stores its data after the # in the URL
+    // So some custom parsing is required
+    const fragment = url.hash;
+    const parts = fragment.split("?", 2);
+
+    const searchParams = new URLSearchParams(parts[1] || "");
+    searchParams.set(
+      "_g",
+      `(time:(from:'${starts_at}',mode:absolute,to:'${ends_at}'))`
+    );
+
+    url.hash = `${parts[0]}?${searchParams.toString()}`;
```

Comment:
```suggestion
    url.hash = `${parts[0]}?${searchParams}`;
```

Expressions in string templates that are not strings are converted to string by calling `toString()` on them